engine/src/property.cpp

```diff
@@ -1237,34 +1237,49 @@ bool MCProperty::resolveprop(MCExecContext& ctxt, Properties& r_which, MCNameRef
         if (!ctxt  . EvalExprAsStringRef(destvar, EE_PROPERTY_BADEXPRESSION, &t_string))
             return false;
 
-        MCScriptPoint sp(*t_string);
+        // AL-2015-08-27: [[ Bug 15798 ]] Parse into index and property name before determining
+        //  if this is a custom property or not (otherwise things like textStyle["bold"] are
+        //  interpreted as custom properties).
+        MCAutoStringRef t_icarray, t_property_name;
+        uindex_t t_offset, t_end_offset;
+        // SN-2014-08-08: [[ Bug 13135 ]] Targetting a custom var should resolve the index AND suppress
+        //  it from the initial name
+        if (MCStringFirstIndexOfChar(*t_string, '[', 0, kMCStringOptionCompareExact, t_offset) &&
+            MCStringLastIndexOfChar(*t_string, ']', UINDEX_MAX, kMCStringOptionCompareExact, t_end_offset) &&
+            t_end_offset == MCStringGetLength(*t_string) - 1)
+        {
+            MCStringCopySubstring(*t_string, MCRangeMake(0, t_offset), &t_property_name);
+            
+            // AL-2015-08-27: [[ Bug 15798 ]] If the index is quoted, we don't want to include the
+            //  quotes in the index name.
+            if (MCStringGetCodepointAtIndex(*t_string, t_offset + 1) == '"' &&
+                MCStringGetCodepointAtIndex(*t_string, t_end_offset - 1) == '"')
+            {
+                t_offset++;
+                t_end_offset--;
+            }
+            
+            MCStringCopySubstring(*t_string, MCRangeMake(t_offset + 1, t_end_offset - t_offset - 1), &t_icarray);
+        }
+        else
+            t_property_name = *t_string;
+        
+        if (*t_icarray != nil)
+        /* UNCHECKED */ MCNameCreate(*t_icarray, t_index_name);
```

Comment:
As I understand it, memory errors will now create an error on the libfoundation error stack. It would therefore be possible to catch them at some level and transmogrify them into script errors. In theory. :confused: